geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/serialization/CollectionWriter.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.feature.serialization
+
+import java.util.{List => JList, Map => JMap}
+
+/** [[DatumReader]]s for reading [[java.util.Map]]s and [[java.util.List]]s.
+ *
+ */
+trait CollectionWriter[Writer] extends PrimitiveWriter[Writer] {
```

Comment:
does it make sense to combine the reader/writer classes, so that the read and write logic is next to each other? or maybe just put them in the same scala file?